docs/develop/go/message-passing.mdx

```diff
@@ -345,13 +351,76 @@ updateHandle, err := temporalClient.UpdateWorkflow(context.Background(), client.
+If it does, the Update is delivered.
+If not, it starts a new Workflow Execution with the supplied ID.
+Upon starting a new Workflow, it immediately processes the Update.
+
+To create a new Update operation, use [`client.NewUpdateWithStartWorkflowOperation`](https://pkg.go.dev/go.temporal.io/sdk/client#NewUpdateWithStartWorkflowOperation).
+The Workflow Id is optional - but when it is specified, it must match the one in `StartWorkflowOptions`.
+It is invalid to set the Run ID since there might not be running Workflow Execution.
+
+Use the `WithStartOperation` option to attach an Update Operation to the [`Client.ExecuteWorkflow`](https://pkg.go.dev/go.temporal.io/sdk/client#Client.ExecuteWorkflow) API call.
+Not all `ExecuteWorkflow` options are allowed with Update-With-Start.
+For example, specifying a Cron Schedule results in an error.
+Refer to the API documentation for further details.
+
+`ExecuteWorkflow` returns once the requested Update wait stage has been reached; or when a provided context times out.
+Choose [Workflow Id Conflict Policy](#workflow-id-conflict-policy) "Use Existing" coupled with an indempotent Update handler by default to allow the code to be executed again, in case of client failure.
+
+You need a [`WorkflowUpdateHandle`](https://pkg.go.dev/go.temporal.io/sdk/client#WorkflowUpdateHandle) to retrieve a result.
+Call `Get` on the Update operation.
+
+`UpdateWithStartWorkflowOperation` can only be executed once.
+Re-using a previously executed operation returns an error from `ExecuteWorkflow`.
+
+The following example shows the creation, configuration, and use of UpdateWithStart:
+
+```go
+ctxWithTimeout, cancel := context.WithTimeout(context.Background(), 15*time.Second)
+defer cancel()
+
+updateOperation := client.NewUpdateWithStartWorkflowOperation(
+    client.UpdateWorkflowOptions{
+        UpdateName:   message.SetLanguageUpdate,
+        WaitForStage: client.WorkflowUpdateStageCompleted,
+    })
+
+startWorkflowOptions := client.StartWorkflowOptions{
+    ID:                       "some-workflow-id",
+    TaskQueue:                "some-task-queue",
+    WithStartOperation:       updateOperation,  // attaches the Update to the Workflow Start
+	WorkflowIDConflictPolicy: enumspb.WORKFLOW_ID_CONFLICT_POLICY_USE_EXISTING,
+}
+we, err := c.ExecuteWorkflow(ctxWithTimeout, startWorkflowOptions, MyWorkflow)
+if err != nil {
+    log.Fatalf("Unable to execute workflow: %v", err)
+}
+
+updateHandle, err := updateOperation.Get(ctxWithTimeout)
+if err != nil {
+    log.Fatalf("Unable to obtain update handle: %v", err)
+}
+
+var previousLang message.Language
+err = updateHandle.Get(ctxWithTimeout, &previousLang)
+if err != nil {
+    log.Fatalf("Unable to obtain update result: %v", err)
+}
+```
+
+For more examples, see [Go sample for early-return pattern](https://github.com/temporalio/samples-go/tree/main/early-return).
+
```

Comment:
I thought about adding some details about how an Update rejection for when a Workflow _is_ started, will always appear in the Workflow History. But I wasn't sure that that is sth users would find very important.

Reply:
I think this is useful info at the point where it makes sense.

Reply:
Talked to Drew and we both think it's not needed.